src/processor/operator/gds_call_shared_state.cpp

```diff
@@ -0,0 +1,42 @@
+#include "processor/operator/gds_call_shared_state.h"
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace processor {
+
+std::vector<NodeSemiMask*> NodeOffsetMaskMap::getMasks() const {
+    std::vector<NodeSemiMask*> masks;
+    for (auto& [_, mask] : maskMap) {
+        masks.push_back(mask.get());
+    }
+    return masks;
+}
+
+FactorizedTable* GDSCallSharedState::claimLocalTable(storage::MemoryManager* mm) {
+    std::unique_lock<std::mutex> lck{mtx};
+    if (availableLocalTables.empty()) {
+        auto table =
+            std::make_shared<processor::FactorizedTable>(mm, fTable->getTableSchema()->copy());
+        localTables.push_back(table);
+        availableLocalTables.push(table.get());
+    }
+    auto result = availableLocalTables.top();
+    availableLocalTables.pop();
+    return result;
+}
+
+void GDSCallSharedState::reclaimLocalTable(FactorizedTable* table) {
```

Comment:
I think 'reclaim' is the wrong word here. I think 'return' is a better word.

Reply:
Done